### ISSUE TITLE
Installing Ocelot

### ISSUE THREAD

**Issue by Kounem** [UncleMeat/gazelle#1]
I'm a little confused as how to install Mifune Ocelot.
Any chance of a quick guide for it as currently I try using the guide https://github.com/WhatCD/Gazelle/wiki/Gazelle-installation-on-Ubuntu-12.04 to install Ocelot and that is not working.
Should I really be using Ocelot 1.0 and adding Mifunes files to it?

**Comment by ValliusDax** [comment]
No, the ocelot version has also been modified to allow for some special features, most notably freeleech.

If you _want_ to try to use the latest WhatCD ocelot and copy the modifications over then go for it, but I highly recommend that you use this one. I'm a CentOS guy, so to compile mifune's ocelot on CentOS 7 then you'd need the following packages:
- gcc 
- gcc-c++ 
- boost 
- boost-thread 
- boost-devel 
- mysql++ 
- mysql++-devel 
- libev 
- libev-devel

Similar packages should be available from your distribution (hopefully). Some Makefile tweaks (mainly for the different include paths) may be required depending on your distribution, here's the CentOS 7 Makefile:

```
CXX=g++
CXXFLAGS=-march=native -O2 -fomit-frame-pointer -fno-ident -fvisibility-inlines-hidden -fvisibility=hidden -Wall -std=c++0x -iquote -Wl,O1 -Wl,--as-needed -I/usr/include/mysql -I/usr/include/mysql++ -I/usr/include/boost -I/usr/include/libev
LDFLAGS=-L/usr/local/lib
LIBS=-lmysqlpp -lboost_system -lboost_thread -lev -lpthread
OCELOT=ocelot
OBJS=$(patsubst %.cpp,%.o,$(wildcard *.cpp))
all: $(OCELOT)
$(OCELOT): $(OBJS)
$(CXX) -pthread $(CXXFLAGS) -o $@ $^ $(LDFLAGS) $(LIBS)
%.o: %.cpp %.h
$(CXX) -c $(CXXFLAGS) -o $@ $<
clean:
rm -f $(OCELOT) $(OBJS)
```

It _should_ compile nicely on Ubuntu, CentOS gives a few warnings which can be ignored. Your mileage may vary, but if you want to chat about it then join us on IRC.

https://webchat.digitalirc.org/?e=web#empornium

**Comment by elaur** [comment]
This ocelot have some problems with doubleseed tokens, sometime is working sometime no. so the only way to fix it was to make a little change in ocelot code where the multiplicate was with *2 i have change with *1 in doubleseed function.

Other problem was on installation with C flags, In Makefile from second line i have removed '-march=native' and than work to compile without errors.

**Comment by ValliusDax** [comment]
I know I'm late in replying, but the doubleseed issue can be resolved easily.

add this at line 737 in worker.cpp

``` c++
       user u;
       u.id = id;
       u.can_leech = 1;
++     u.pfl = 0;
       users_list[passkey] = u;
```

**Comment by mohawk123** [comment]
Anyone still look at this? looking for some help setting this up. no real guide on it

**Comment by Empornium** [comment]
This repository is deprecated, the currently maintained code is now at:

https://github.com/Empornium/Luminance 
https://github.com/Empornium/Radiance
